<a href="https://colab.research.google.com/github/Tirtayuda12/Machine_Learning-Telu/blob/WEEK-1/Decision_Tree_Tirtayuda_Munggarana.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Name : Tirtayuda Munggarana

NIM : 1103202108

Dataset : melb_house.csv

Overview

Import Library

In [1]:
import pandas as pd #import modul pandas
from sklearn.model_selection import train_test_split #import fungsi data latih dan uji dari sklearn
from sklearn.tree import DecisionTreeRegressor #impot dari modul tree
from sklearn.metrics import mean_squared_error #import fungsi MSE
from sklearn.impute import SimpleImputer #import fungsi simple imputer

Membaca file csv

In [2]:

# Baca data dari file CSV
url = "/content/melb_data.csv"
data = pd.read_csv(url)

Menampilkan data

In [3]:
# Tampilkan lima baris pertama dari data
print(data.head())

   Unnamed: 0      Suburb           Address  Rooms Type      Price Method  \
0           1  Abbotsford      85 Turner St      2    h  1480000.0      S   
1           2  Abbotsford   25 Bloomburg St      2    h  1035000.0      S   
2           4  Abbotsford      5 Charles St      3    h  1465000.0     SP   
3           5  Abbotsford  40 Federation La      3    h   850000.0     PI   
4           6  Abbotsford       55a Park St      4    h  1600000.0     VB   

  SellerG       Date  Distance  ...  Bathroom  Car  Landsize  BuildingArea  \
0  Biggin  3/12/2016       2.5  ...       1.0  1.0     202.0           NaN   
1  Biggin  4/02/2016       2.5  ...       1.0  0.0     156.0          79.0   
2  Biggin  4/03/2017       2.5  ...       2.0  0.0     134.0         150.0   
3  Biggin  4/03/2017       2.5  ...       2.0  1.0      94.0           NaN   
4  Nelson  4/06/2016       2.5  ...       1.0  2.0     120.0         142.0   

   YearBuilt  CouncilArea  Lattitude Longtitude             Regionna

Menggunakan one hot encoding

In [4]:
# One-hot encode categorical variables
X = pd.get_dummies(data.drop('Rooms', axis=1))

# Fitur
y = data['Rooms']  # Target

Split Dataset

In [5]:
# Bagi data menjadi data latih dan data uji
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

Membuat objek simple imputer

In [6]:
# Tangani nilai yang hilang menggunakan SimpleImputer
imputer = SimpleImputer(strategy='mean')
imputer.fit(X_train)
X_train_imputed = imputer.transform(X_train)
X_test_imputed = imputer.transform(X_test)

Membuat model Decision Tree

In [7]:
# Buat model Decision Tree
model = DecisionTreeRegressor(random_state=42)

Proses pelatihan model

In [8]:
# Latih model pada data latih
model.fit(X_train_imputed, y_train)

DecisionTreeRegressor(random_state=42)

Melakukan prediksi

In [9]:
# Lakukan prediksi pada data uji
y_pred = model.predict(X_test_imputed)

Evaluasi Model

In [10]:
# Evaluasi model
mse = mean_squared_error(y_test, y_pred)
print("Mean Squared Error:", mse)

Mean Squared Error: 0.16956521739130434


Kesimpulan

Output "Mean Squared Error: 0.16956521739130434" adalah hasil evaluasi kinerja model regresi pohon keputusan menggunakan metrik Mean Squared Error (MSE). Nilai MSE yang lebih kecil menunjukkan bahwa model memiliki kinerja yang lebih baik.

Namun, untuk memberikan penilaian yang lebih lengkap tentang kinerja model, penting untuk membandingkan nilai MSE dengan model lain atau mempertimbangkan metrik evaluasi lainnya juga, tergantung pada konteks masalah dan tujuan analisisnya.